src/main/java/org/reactome/release/orthopairs/UniProtGeneNamesRetriever.java

```diff
@@ -0,0 +1,178 @@
+package org.reactome.release.orthopairs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.ac.ebi.kraken.interfaces.uniprot.Gene;
+import uk.ac.ebi.uniprot.dataservice.client.Client;
+import uk.ac.ebi.uniprot.dataservice.client.QueryResult;
+import uk.ac.ebi.uniprot.dataservice.client.ServiceFactory;
+import uk.ac.ebi.uniprot.dataservice.client.exception.ServiceException;
+import uk.ac.ebi.uniprot.dataservice.client.uniprot.UniProtComponent;
+import uk.ac.ebi.uniprot.dataservice.client.uniprot.UniProtQueryBuilder;
+import uk.ac.ebi.uniprot.dataservice.client.uniprot.UniProtService;
+import uk.ac.ebi.uniprot.dataservice.query.Query;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.*;
+
+public class UniProtGeneNamesRetriever {
+
+    private static final Logger logger = LogManager.getLogger();
+    private static final int MAX_UNIPROT_BATCH_QUERY_SIZE = 250;
+
+    /**
+     * Queries the UniProt mapping service through their Java API library. All Uniprot accession IDs are taken from the Panther
```

Comment:
My nitpick for the day (I can stop doing this if it's getting annoying :D) - Uniprot can be changed to have a captialized P :).  I actually would have preferred if it were lowercase in the name as it can be confusing with camelcase naming.